### DISCUSSION TITLE
Line Clamping not working as expected

### DISCUSSION THREAD

**Discussion by dev-gush** [tailwindlabs/tailwindcss#12095] [Help]
I am using `line-clamp-3` to truncate a text after 3 lines but it adds a 4th line after the ellipsis (`...`), like so -

![image](https://github.com/tailwindlabs/tailwindcss/assets/128897000/906772a3-b9df-4827-a382-5fec03b16277)

my code is as follows - 

```
<div className="p-[24px] text-[#0C0C30] text-[14px] tracking-[0.2px] line-clamp-3">A curated, cleaned lead list is sent within a few hours for review - make clean-up and analytics suggestions as many times as you want! A curated, cleaned lead list is sent within a few hours for review - make clean-up and analytics suggestions as many times as you want!</div>
```

**What version of Tailwind CSS are you using?**

3.3.2

**What build tool (or framework if it abstracts the build tool) are you using?**

```
"postcss": "8.4.23",
"next": "13.3.2"
```

**What version of Node.js are you using?**

v18.16.0

**What browser are you using?**

Edge

**What operating system are you using?**

macOS

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.

**Comment by adamwathan** [comment]
Hey! This is just because of how padding interacts with line-clamp in CSS — since you have padding on the element, the extra line of text is visible in the padding. Weird way to build the CSS feature IMO but it is what it is.

Solution is put the padding on a wrapper:

https://play.tailwindcss.com/sVGRF67SDE?size=430x773

**Reply by kevinashworth** [reply]
I was also seeing `line-clamp-X` not working when `block` was in `className`, too.

**Reply by semanticist21** [reply]
The soultion seems not intuitive for me.. but it works great.
